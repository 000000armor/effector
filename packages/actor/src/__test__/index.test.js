//@flow

import {createActor} from '..'

test('smoke', () => {
 const actor = createActor([
  {
   foo: 'bar',
   count: 0,
  },
  {
   tag: 'computed',
   value: 3,
  },
 ])
 actor.pushMessage('hello')
 actor.pushMessage('how r u')
 expect(actor.state).toMatchSnapshot()
})

test('send message', async() => {
 const actor = createActor([
  {
   foo: 'bar',
   count: 0,
  },
  {
   tag: 'computed',
   value: 3,
  },
 ])
 expect(actor.state).toMatchSnapshot()

 await actor.sendMessage('hello')
 await actor.sendMessage('how r u')

 expect(actor.state).toMatchSnapshot()
})

test('async getState', async() => {
 const actor = createActor([
  {
   foo: 'bar',
   count: 0,
  },
  {
   tag: 'computed',
   value: 3,
  },
 ])

 const state = await actor.getState()
 expect(state).toMatchSnapshot()
})

test('after', async() => {
 const fn = jest.fn()
 const fn1 = jest.fn()
 const actor = createActor([
  {
   foo: 'bar',
   count: 0,
  },
  {
   tag: 'computed',
   value: 3,
  },
 ])
 actor.after(
  (state, e) => {
   if (e !== 'inc') return
   return state[0]
  },
  async stateA => {
   stateA.count += 1
   fn(stateA.count)
   console.log(stateA)
  },
 )
 actor.after(
  (state, e) => {
   if (e !== 'print count') return
   if (state[0].count > 0) return state[0].count
  },
  async count => {
   fn1(count)
   console.log(count)
  },
 )
 await actor.sendMessage('inc')
 await actor.readIncoming()
 await actor.sendMessage('inc')
 await actor.sendMessage('print count')
 await actor.readIncoming()
 expect(actor.state.state[0].count).toBe(2)
})
