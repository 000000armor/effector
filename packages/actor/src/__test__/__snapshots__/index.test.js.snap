// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`async getState 1`] = `
Array [
  Object {
    "count": 0,
    "foo": "bar",
  },
  Object {
    "tag": "computed",
    "value": 3,
  },
]
`;

exports[`send message 1`] = `
State {
  "events": Set {},
  "state": Array [
    Object {
      "count": 0,
      "foo": "bar",
    },
    Object {
      "tag": "computed",
      "value": 3,
    },
  ],
}
`;

exports[`send message 2`] = `
State {
  "events": Set {
    "hello",
    "how r u",
  },
  "state": Array [
    Object {
      "count": 0,
      "foo": "bar",
    },
    Object {
      "tag": "computed",
      "value": 3,
    },
  ],
}
`;

exports[`smoke 1`] = `
State {
  "events": Set {
    "hello",
    "how r u",
  },
  "state": Array [
    Object {
      "count": 0,
      "foo": "bar",
    },
    Object {
      "tag": "computed",
      "value": 3,
    },
  ],
}
`;
